src/bin/cargo.rs

```diff
@@ -166,9 +169,54 @@ fn execute(flags: Flags, config: &Config) -> CliResult<Option<()>> {
         // For all other invocations, we're of the form `cargo foo args...`. We
         // use the exact environment arguments to preserve tokens like `--` for
         // example.
-        _ => env::args().collect(),
+        _ => {
+            let mut default_alias = HashMap::new();
+            default_alias.insert("b", "build".to_string());
+            default_alias.insert("t", "test".to_string());
+            default_alias.insert("r", "run".to_string());
+            let mut args: Vec<String> = env::args().collect();
+            if let Some(new_command) = default_alias.get(&args[1][..]){
+                args[1] = new_command.clone();
+            }
+            args
+        }
     };
 
+    match config.get_string(&format!("alias.{}", args[1])){
+        Ok(value) =>{
+            match value {
+                Some(record) => {
+                    // Group quoted items together
+                    let regex = Regex::new("(\"[^\"]*\"|[^\"]+)").unwrap();
```

Comment:
Oh sorry, what I meant here was something like:

* If the configuration is a string, then we just split on whitespace
* We also accept a configuration of a list of strings, in which case we do not splitting. In that case I think we can avoid using a regex here